python/dolfinx/fem/forms.py

```diff
@@ -631,3 +631,27 @@ def create_form(
         msh._cpp_object,
     )
     return Form(f, form.ufcx_form, form.code)
+
+
+def compute_jacobian(
+    F: typing.Union[ufl.Form | list[ufl.Form]],
+    u: typing.Union[Function | list[Function]],
+    du: typing.Optional[typing.Union[ufl.Argument, list[ufl.Argument]]] = None,
+) -> typing.Union[ufl.Form, list[list[ufl.Form]]]:
+    """Compute the Jacobian :math:`J = \\frac{\\partial F}{\\partial u}[\\delta u]`.
```

Comment:
It's not clear from the docs what is 'computed'. It should say something like it computes the derivative of a UFL form to generate a UFL form for the Jacobian.

Need to document the arguments.

Reply:
I'm going to review and edit all of the documentation in due course. Many issues, also with consistency with respect to `LinearProblem`.